claim/utils.py

```diff
@@ -8,6 +10,8 @@ def process_child_relation(user, data_children, claim_id, children, create_hook)
     for data_elt in data_children:
         claimed += data_elt['qty_provided'] * data_elt['price_asked']
         elt_id = data_elt.pop('id') if 'id' in data_elt else None
+        if __check_if_maximum_amount_overshoot(data_children, children):
```

Comment:
Shouldn't this be outside for loop? Seems you're doing N² validation with this.

Reply:
Moved outside the loop.